src/API/Site/Controllers/Ads/BudgetRecommendationController.php

```diff
@@ -52,47 +52,89 @@ public function __construct( RESTServer $rest_server, BudgetRecommendationQuery
 	 */
 	public function register_routes(): void {
 		$this->register_route(
-			'ads/campaigns/budget-recommendation/(?P<country_code>\\w{2})',
+			'ads/campaigns/budget-recommendation',
 			[
 				[
 					'methods'             => TransportMethods::READABLE,
 					'callback'            => $this->get_budget_recommendation_callback(),
 					'permission_callback' => $this->get_permission_callback(),
+					'args'                => $this->get_collection_params(),
 				],
 				'schema' => $this->get_api_response_schema_callback(),
 			]
 		);
 	}
+
+	/**
+	 * Get the query params for collections.
+	 *
+	 * @return array
+	 */
+	public function get_collection_params(): array {
+		return [
+			'context'       => $this->get_context_param( [ 'default' => 'view' ] ),
+			'country_codes' => [
+				'type'              => 'array',
+				'sanitize_callback' => $this->get_country_code_sanitize_callback(),
+				'validate_callback' => $this->get_country_code_validate_callback(),
+				'items'             => [
+					'type' => 'string',
+				],
+			],
+		];
+	}
+
 	/**
 	 * @return callable
 	 */
 	protected function get_budget_recommendation_callback(): callable {
 		return function( Request $request ) {
-			$country        = strtoupper( $request->get_param( 'country_code' ) );
-			$currency       = $this->middleware->get_ads_currency();
-			$recommendation = $this
+			$country_codes = $request->get_params()['country_codes'];
```

Comment:
Why not just `get_param`? One of the warnings that appears is `Undefined index: country_codes` if we send a request without country_codes (see comment on required collection parameter).

```suggestion
			$country_codes = $request->get_param( 'country_codes' );
```

Reply:
Ah my bad, I replaced it from `get_query_params` and kept the `params` so it became `get_params` 😓. I should check the log messages to see if there are any warnings. Thanks for pointing this out!

Updated in 37e5212.